plugins/amazonq/codetransform/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codemodernizer/ArtifactHandler.kt

```diff
@@ -275,8 +277,13 @@ class ArtifactHandler(private val project: Project, private val clientAdaptor: G
     /**
      * Opens the built-in patch dialog to display the diff and allowing users to apply the changes locally.
      */
-    internal suspend fun displayDiffUsingPatch(patchFile: VirtualFile, totalPatchFiles: Int, diffDescription: PatchInfo?, jobId: JobId,
-                                               source: CodeTransformVCSViewerSrcComponents) {
+    internal fun displayDiffUsingPatch(
+        patchFile: VirtualFile,
+        totalPatchFiles: Int,
+        diffDescription: PatchInfo?,
+        jobId: JobId,
+        source: CodeTransformVCSViewerSrcComponents,
+    ) {
         runInEdt {
```

Comment:
I think the better way to do this is to use `withContext(EDT) {` here, make this function `suspend`, and then get rid of the `projectCoroutineScope(project).launch {` below on line 312, but @rli can correct me if I'm wrong